### ISSUE TITLE
active_month doesn't seem to be reliably reported

### ISSUE THREAD

**Issue by markrprior** [mastodon/mastodon#23796]
### Steps to reproduce the problem

I have a script that walks the fediverse counting and reporting on stuff and I have noticed a number of systems that report an active_month count larger than the user count. Perhaps this is expected but I find it strange. As an example look at zbrx.org (not I am not affiliated with zbrx.org but it's an example of the problem.
If you look at the nodeinfo record you see

```
{
          'openRegistrations' => bless( do{\(my $o = 1)}, 'JSON::PP::Boolean' ),
          'usage' => {
                       'localPosts' => 401,
                       'users' => {
                                    'activeHalfyear' => 11,
                                    'activeMonth' => 11,
                                    'total' => 4
                                  }
                     },
          'protocols' => [
                           'activitypub'
                         ],
          'metadata' => {},
          'version' => '2.0',
          'software' => {
                          'name' => 'mastodon',
                          'version' => '4.0.2'
                        },
          'services' => {
                          'inbound' => [],
                          'outbound' => []
                        }
        };
```

So it's claiming 4 users but 11 active both for the last month and year. Looking at the Mastodon instance API for the contact account it would seem to have been created recently
```
'created_at' => '2023-01-31T00:00:00.000Z'
```
so maybe this problem relates to that, some initialisation problem or dealing with incomplete data.
Looking at the activity log
```
[{"week":"1677038999","statuses":"6","logins":"0","registrations":"0"},{"week":"1676434199","statuses":"220","logins":"9","registrations":"0"},{"week":"1675829399","statuses":"43","logins":"4","registrations":"0"},{"week":"1675224599","statuses":"45","logins":"5","registrations":"4"},{"week":"1674619799","statuses":"0","logins":"0","registrations":"0"},{"week":"1674014999","statuses":"0","logins":"0","registrations":"0"},{"week":"1673410199","statuses":"0","logins":"0","registrations":"0"},{"week":"1672805399","statuses":"0","logins":"0","registrations":"0"},{"week":"1672200599","statuses":"0","logins":"0","registrations":"0"},{"week":"1671595799","statuses":"0","logins":"0","registrations":"0"},{"week":"1670990999","statuses":"0","logins":"0","registrations":"0"},{"week":"1670386199","statuses":"0","logins":"0","registrations":"0"}]
```
We potentially see the 4 users created but it's not clear how the activity is derived.

### Expected behaviour

I'm expecting active_month to be equal to or less than the user count for most instances. Instances closing down and deleting accounts would be the exception.

### Actual behaviour

A number of instances report an active_month count larger than the user count

### Specifications

Seems to have been around for some time since it's not specific to a version.

dlyang.tk -- mastodon [3.1.2] 10 47
masto.yttrx.com -- mastodon [4.1.0] 114 181
norcal.social -- mastodon [4.1.0] 961 1011
okapi.website -- mastodon [3.1.3] 48 185
ourislandgeorgia.net -- mastodon [3.1.4<wbr>+glitch] 13 266
space-pirates.org -- mastodon [4.0.2] 63 342
unperson.us -- mastodon [3.1.3] 7 61
www.paradiseve.com -- mastodon [3.2.1] 44 205
zbrx.org -- mastodon [4.0.2] 4 11

**Comment by ClearlyClaire** [comment]
The active user count is a mostly-accurate approximation of the number of different accounts that were found to be logged in at least once (including right after account creation) during that time frame. This number does not go down when an account “active” in this time frame gets deleted.

**Comment by markrprior** [comment]
@ClearlyClaire while I understand it would be hard to deal with the accounting associated with an active account being deleted this doesn't appear to be the situation, at least with zbrx.org where it appears to be a new instance with 4 users but the active count is 11. The difference in numbers for space-pirates.org is crazy. I wouldn't have thought that 4/5 of the accounts that were active in the last month have suddenly been deleted although I haven't asked the admin that question.

**Comment by ClearlyClaire** [comment]
These servers have registrations open. I think that the active users may also track accounts that have not confirmed their email address. Gonna check that.

**Comment by ClearlyClaire** [comment]
I can confirm that registering an account without confirming the mail address bumps the active user count without bumping the total user count. This probably accounts for most of the discrepancy, and should be addressed by #23803

**Comment by vmstan** [comment]
Fixed in #23803, closing.
